api/app/controllers/transcript.controller.js

```diff
@@ -1,23 +1,61 @@
+const authConfig = require("../config/auth.config.js");
 const db = require("../models");
 const Transcript = db.transcripts;
 const Op = db.Sequelize.Op;
+const crypto = require('crypto');
 
 // Create and Save a new Transcript
 exports.create = (req, res) => {
   // Validate request
   if (!req.body.content) {
-    //FIXME: Include original content check in if condition
     res.status(400).send({
-      message: "Content can not be empty!"
+      message: "Content cannot be empty!"
+    });
+    return;
+  }
+  
+  else if (!req.body.originalContent) {
+    res.status(400).send({
+      message: "Original Content cannot be empty!"
+    });
+    return;
+  }
+
+  // add authentication token to prevent dos attacks
+  else if (!req.body.authToken) {
+    res.status(400).send({
+      message: "Auth token cannot be empty!"
     });
     return;
   }
 
+
+  const verifyToken = (token) => {
+
+    return token === authConfig.TOKEN;
+  }
+
+  const generateHash = () => {
+
+    const oc = req.body.originalContent
+    const hashParams = oc.title + oc.media + oc.date
+    const transcriptHash = crypto.createHash('sha256').update(hashParams).digest('base64');
```

Comment:
It could be a security risk using a static token, also since we are including hashParams (oc.title+oc.media+oc.date) the token will be different for each payload. Would the static TOKEN work?
We can have a shared secret then hash it with that secret and the payload parameters then compare the hashes.

```js
const key = crypto.createHash('sha256').update(secret).digest();
const hashParams = oc.title + oc.media + oc.date
const transcriptHash = crypto.createHash('sha256', key).update(hashParams).digest('base64');
```

Reply:
so i think we are confusing two different things here. the `transcriptHash` and the `authToken` .For the transcript hash, it should be unique for every transcript to be double posting. For the auth token .I initially had a naive implementation of a static token ,but given what we've discussed on the call ,am going to implement a hash of a shared secret and have it passed through the header. Incidentally, that would also fix #1

Reply:
isn't there a possibility that the media source in the JSON payload could be null? in that case do we just dismiss the transcript?

Reply:
if that's the case, I'd do a match on title + first 5 words of the body. I think a hash is overkill given that whitespace and casing with change all the hashes and I'd prefer to avoid the dependency.